intermine/query_manager.py

```diff
@@ -0,0 +1,198 @@
+
+    r = requests.get(x)
+    dict = yaml.load(r.text)
+    link = dict["instance"]["url"] + "/service/user/queries?token=" + token
+
+    r = requests.get(link)
+    dict = yaml.load(r.text)
+    count = 0
+    for key in dict['queries'].keys():
+        if(name == key):
+            count = count + 1
+            print("Columns:")
+            for i in range(len(dict['queries'][name]['select'])):
+                print(dict['queries'][name]['select'][i])
+    if(count == 0):
+        print("No such query available")
+
+
+def delete_query(name):
+    """
+    A function that deletes a given query
+    ================================================
+    example:
+
+        >>>from intermine import query_manager as qm
+        >>>qm.delete_query('queryName')
+        <deletes the query whose name is 'queryName' from user's account>
+
+    """
+    x = "http://registry.intermine.org/service/instances/" + mine
+    r = requests.get(x)
+    dict = yaml.load(r.text)
+
+    y = dict["instance"]["url"] + "/service/user/queries?token=" + token
+    r = requests.get(y)
+    z = yaml.load(r.text)
+    count = 0
+    for key in z['queries'].keys():
+        if (key == name):
+            count = count + 1
+    if(count == 0):
+        print("No such query available")
+
+    else:
+        link = dict["instance"]["url"] + "/service/user/queries/" + name +\
+         "?token=" + token
+        requests.delete(link)
+
+    '''
+    x = "http://registry.intermine.org/service/instances/" + mine
+    r = requests.get(x)
+    dict = yaml.load(r.text)
+    link = dict["instance"]["url"] + "/service/user/queries/" + name +\
+     "?token=" + token
+    requests.delete(link)
+    '''
+
+def xml_to_link(xml):
+    """
+    Converts xml to the link we desire while posting queries
+    ================================================
+    example:
+        >>>xml_to_link('<query></query>')
+        '%3Cquery%3E%3C%2Fquery%3E'
+    """
+    count = 0
+    link = []
+    for i in xml:
+        if (i == "<"):
+            link.append('%3C')
+        elif (i == ">"):
+            link.append('%3E')
+        elif (i == "="):
+            link.append('%3D')
+        elif (i == " "):
+            link.append('%20')
+        elif (i == '"'):
+            link.append('%22')
+        elif (i == "/"):
+            link.append('%2F')
+        else:
+            link.append(i)
+    link = "".join(link)
+    return link
+
+def post_query(xml):
+    """
+    A function to post a query(in the form of string containg xml) to a user account
+    ================================================
+    example:
+
+        >>>qm.post_query('<query name="" model="genomic" view="Gene.length Gene.symbol" \
+           longDescription="" sortOrder="Gene.length asc"></query>')
+    """
+    x = "http://registry.intermine.org/service/instances/" + mine
+    r = requests.get(x)
+    dict = yaml.load(r.text)
+    name = []
+    for i in range(13,len(xml)):
```

Comment:
What is 13? A magic number!

https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants

Reply:
No, actually in every xml string, the 13th element will be the one where name starts.
example: '<query name="xxx" model=.....
< = 0th position
q = 1st position
u = 2nd position
....
" = 12th position
x = ***13th position***

Reply:
Maybe this is clearer explanation as to why this is a bad thing:

https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad

Reply:
Oh right! This seems better, will do the necessary change